src/ToggleButton/ToggleButton.js

```diff
@@ -0,0 +1,212 @@
+        }
+      } else {
+        values.splice(ind, 1);
+        indexes.splice(ind, 1);
+      }
+    } else {
+      if (this.props.exclusive) {
+        values = [];
+        indexes = [];
+      }
+      values.push(option.value);
+      indexes.push(option.index);
+    }
+
+    if (indexes.length > 0) {
+      active = true;
+    }
+
+    this.setState({
+      values,
+      active,
+      selectedOptions: indexes,
+    });
+
+    if (indexes.indexOf(option.index) > -1) {
+      if (option.onSelect) {
+        option.onSelect(option.value);
+      }
+    } else if (option.onDeselect) {
+      option.onDeselect(option.value);
+    }
+  };
+
+  render() {
+    const {
+      children,
+      active: activeProp,
+      selectedOptions,
+      values,
+      exclusive,
+      classes,
+      className: classNameProp,
+      toggleIcons,
+      ...other
+    } = this.props;
+
+    const active = this.state.active;
+
+    const className = classNames(
+      classes.root,
+      {
+        [classes.active]: active && !toggleIcons,
+        [classes.toggleIcon]: toggleIcons,
+      },
+      classNameProp,
+    );
+
+    const options = this.getOptions().map((option, index) => {
+      // warning(option.type && option.type.muiName === 'Option',
+      //   `Material-UI: ToggleButton only accepts Option Components as children.
+      //   Found ${option.type.muiName || option.type}
+      //   as child number ${index + 1} of ToggleButton`);
+      //
+      // if(!option.props.children){
+      //   warning(option.props.value !== undefined,
+      //     `Material-UI: ToggleButton value prop has been passed, but Option ${index}
+      //   does not have a value prop. Needs value if ToggleButton is going
+      //   to be a controlled component.`);
+      // }
+
+      const selected = this.getSelected(index);
+      let divider = false;
+
+      if (selected && !toggleIcons && index > 0) {
+        if (this.getSelected(index - 1)) {
+          divider = true;
+        }
+      }
+
+      return cloneElement(option, {
+        key: index,
+        index,
+        divider,
+        selected,
+        noBackground: toggleIcons,
+        onClick: this.handleOptionClick,
+      });
+    });
+
+    return (
+      <div className={className} {...other}>
+        {options}
+      </div>
+    );
+  }
+}
+
+ToggleButton.propTypes = {
+  /**
+   * Determines whether or not the 'ToggleButton' component is selected.
+   */
+  active: PropTypes.bool,
+  /**
+   * Should be used to pass `Option` components.
+   */
+  children: PropTypes.node,
+  /**
+   * Useful to extend the style applied to components.
+   */
+  classes: PropTypes.object.isRequired,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * Determines whether or not the ToggleButton is multivalued.
+   */
+  exclusive: PropTypes.bool,
+  /**
+   * Indexes of the currently selected options on the 'ToggleButton'.
+   */
+  selectedOptions: PropTypes.arrayOf(PropTypes.number),
```

Comment:
What about calling that property `value`?

Reply:
We use the `value`/onChange` couple as much as we can for controlling components. How doesn't that match what's store?